presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -137,6 +137,20 @@ public void testInnerInequalityJoinWithEquiJoinConjuncts()
     }
 
     @Test
+    public void testLeftConvertedToInnerInequalityJoinNoEquiJoinConjuncts()
+            throws Exception
+    {
+        assertPlan("SELECT 1 FROM orders o LEFT JOIN lineitem l ON o.orderkey < l.orderkey WHERE l.orderkey IS NOT NULL",
+                anyTree(
+                        filter("O_ORDERKEY < L_ORDERKEY",
+                                join(INNER, ImmutableList.of(), Optional.empty(),
```

Comment:
Actually it is converted to cross join. So no prunning should appear on `JoinNode` level.